Packs/MicrosoftGraphIdentityandAccess/Scripts/MSGraphIdentityBlockExternalIPWithCAPolicy/MSGraphIdentityBlockExternalIPWithCAPolicy.py

```
@@ -0,0 +1,297 @@
+            "clientAppTypes": ["all"],
+            "locations": {"includeLocations": [named_location_id]},
+        },
+        "grantControls": {"operator": "OR", "builtInControls": ["block"]},
+        "sessionControls": {
+            "applicationEnforcedRestrictions": None,
+            "cloudAppSecurity": None,
+            "persistentBrowser": None,
+            "signInFrequency": None,
+            "disableResilienceDefaults": False,
+        },
+    }
+    demisto.debug(f"Attempting to create CA policy with JSON: {json.dumps(policy_json, indent=2)}")
+    _execute_command_and_handle_error(
+        "msgraph-identity-ca-policy-create", {"policy": json.dumps(policy_json)}, "Failed to create CA policy"
+    )
+    demisto.debug(f"Successfully created Conditional Access policy '{policy_name}'.")
+
+
+def block_external_ip_with_ca_policy_main_logic(ip: str, named_location_name: str, policy_name: str) -> str:
+    """
+    Orchestrates the blocking of an external IP address using Azure Conditional Access policies.
+    This function checks if the IP is private, retrieves or updates a named IP location in Azure AD,
+    and then either updates an existing Conditional Access policy or creates a new one to block
+    access from the specified IP.
+    Args:
+        ip (str): The external IP address to block.
+        named_location_name (str): The name of the Azure AD named IP location to use/create.
+        policy_name (str): The name of the Conditional Access policy to use/create.
+    Returns:
+        str: A message indicating the successful processing and blocking status of the IP.
+    Raises:
+        DemistoException: If the input IP is missing, private, or if any Azure AD operation fails.
+    """
+    result_message = ""
+    if not ip:
+        raise DemistoException("Missing required argument: 'ip'.")
+    if is_private_ip(ip):
+        raise DemistoException(f"The IP {ip} appears to be internal/private and will not be blocked.")
```

Do we want to raise an exception? Maybe just log and return a human readable? This can break the PB.
Also think of other error handling if there is any case that you want the script to proceed and not fail.